apps/utils.py

```diff
@@ -136,3 +140,18 @@ def serialize(self, queryset, *, stream=None, fields=None, use_natural_foreign_k
                 self.first = False
         self.end_serialization()
         return self.getvalue()
+
+
+def send_templated_email(template=None, subject=None, context=None, sender=None, recipients=None):
+    if not template or not subject or not recipients:
+        raise ValidationError
```

Comment:
It seems that the only time this is hit is if there was an internal error. I would maybe change this to maybe a 503 error, so there is no confusion for what caused the error. I would check with @ajhodges about his thoughts on it though.

Reply:
I like that there is input validation in this function - I'm not sure if its really a ValidationError - that implies that the User did something wrong. It's not a 503 - 503's mean that a service was unavailable (like if the SMTP server was down or something). `not template or not subject or not recipients` implies that something went wrong in the serializer. If the serializer has an issue, it should throw an exception. This function can probably just 'trust' the input that its given, but it should make `template`, `subject` and `recipients` required parameters.

Reply:
@ClovisDj as mentioned above I think I prefer making those parameters to `send_templated_mail` required instead of throwing `ValueError`/`TypeError` exceptions. We can trust the output of the serializer to not return `None`. I also don't think we need to default those parameters to `None` if they will always throw an exception if the default value is used.